app/Http/Controllers/LibraryController.php

```diff
@@ -9,29 +9,38 @@
 use App\Models\Library;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
-
+use Illuminate\Support\Facades\Session;
 class LibraryController extends Controller
 {
     public function library()
 {
-
     $user = Auth::user();
-    if(Auth::check()){
-        $userBookIDs = Library::where('user_ID', Auth::user()->user_ID)->pluck('book_ID')->toArray();
-
-        // Count the number of book IDs
-        $count = count($userBookIDs);
-
-        // Get all book objects for the user
-        $books = Books::whereIn('book_ID', $userBookIDs)->get();
-
-        // Output the count
+    // Get all book objects for the user
+    $books = $user->booksInLibrary;
+    return view('library', compact('books'));
 
-
-        return view('library', compact('books'));
-
-    } else{
-        return view('please_login');
-    }
+    
 }
+public function addToLibrary(Request $request)
+    {
+        
+        $book_id = $request->input('book_ID');
+        // dd($book_ID);
+
+        $user = Auth::user();
+        // Check if the book is already in the user's library
+        $library = $user->library;
+        $bookIDs = $library->pluck('book_ID')->toArray();
+        $isInLibrary = in_array($book_id, $bookIDs);   
+        if ($isInLibrary) {
```

Comment:
if it is already in the library, add it again? If it is not, then don't add it? 

I believe this is not the intented behaviour